docs/proposals/20231123-enhance-mid-tier-resource.md

```diff
@@ -0,0 +1,187 @@
+---
+title: Enhance Mid-tier resources
+authors:
+  - "@j4ckstraw"
+  - "@jiasheng55"
+reviewers:
+  - "@zwzhang0107"
+  - "@hormes"
+  - "@eahydra"
+  - "@FillZpp"
+  - "@jasonliu747"
+creation-date: 2023-11-23
+last-updated: 2023-11-28
+status: implementable
+see-also:
+  - "/docs/proposals/20230613-node-prediction.md"
+---
+
+# Enhance Mid-tier resources
+
+## Table of Contents
+
+<!--ts-->
+- [Enhance Mid-tier resources](#enhance-mid-tier-resources)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Design Principles](#design-principles)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+      - [Cgroup basic configuration](#cgroup-basic-configuration)
+      - [QoS policy](#qos-policy)
+      - [Node QoS](#node-qos)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+  - [Implementation History](#implementation-history)
+<!--te-->
+
+## Summary
+
+The *Mid-tier resources* is proposed to both improve the node utilization and avoid overloading, which rely on [node prediction](https://github.com/koordinator-sh/koordinator/blob/main/docs/proposals/20230613-node-prediction.md). 
+
+While *node prediction* clarify how the Mid-tier resources are calculated with the prediction, and this proposal will clarify Mid-tier cgroup and QoS design, suppress and eviction policy.
+
+## Motivation
+
+Here I want to explain some concepts:
+1. koord-QoS
+
+Quality of Service, we assue the same QoS level has similar operating performance, operating quality.
+
+2. koord-priority
+
+Scheduling priority, high priority can preempt low priority by default.
+
+3. Resource Type
+
+We have four resource type now, prod, mid, batch and free resource
+resource type care about whether the resource is oversold, whether it is stable, which affects pod eviction.
+
+koordinator bind koord-priority and resource type, different priority has different resource type.
+
+This proposal introduce Mid+LS and Mid+BE fill the gap in Prod+LS and Batch+BE
+meet the requirements of different types of tasks.
+
+### Goals
+
+- Clarify Mid-tier cgroup and QoS
+- Clarify Mid-tier suppress and eviction policy
+
+### Non-Goals
+
+- Replace Batch-tier resources
+- Add new QoS type
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+There are low-priority online-service tasks, which performance requirements is same as Prod+LS while it do not want to be suppressed but can tolerate being evicted, when the machine usage spike. 
```

Comment:
If the Mid pods are allowed to allocate the Prod unallocated resources, the Prod apps can be affected since `ProdPeak + MidAllocated > NodeAllocatable` is possible. So to avoid the affection, there should be a design about when and how we let the Prod pods preempt/evict Mid pods when they want to win back the resources that Prod unallocated but Mid allocated.

Reply:
Yes, Because prod pods use cpu/memory resource, while mid pods use mid-cpu/mid-memory, so prod can't preempt mid directly. 
How about preempt/evict by `mid-allocated / mid-allocatable`.